apps/wallet-dashboard/components/Dialogs/Staking/StakeDialog.tsx

```diff
@@ -77,17 +125,26 @@ function StakeDialog({
 
     const validators = Object.keys(rollingAverageApys ?? {}) ?? [];
 
-    function handleNext(): void {
-        setStep(Step.EnterAmount);
-    }
-
     function handleBack(): void {
-        setStep(Step.SelectValidator);
+        setView?.(StakeDialogView.SelectValidator);
```

Comment:
If I open the dialog from New stake flow, this is the expected behavior, but if I open the dialog from the details of a stake, it does not

https://www.awesomescreenshot.com/video/33975240?key=3b159a5665765330f185fb89d1f4fa30

In this example, instead of returning to the details view, the popup is closed. Maybe we could open a new issue to better handle the back function, instead of fixing it in this PR. wdyt?

Reply:
thanks ❤️